Makefile

```diff
@@ -1,15 +1,17 @@
-.PHONY: compile clean test rel run docker-build docker-test docker-run
+.PHONY: compile clean test rel run grpc docker-build docker-test docker-run
+
+grpc_services_directory=src/grpc/autogen
 
 REBAR=./rebar3
 
-compile: |
-	$(REBAR) compile 
+compile: | $(grpc_services_directory)
```

Comment:
on miner we removed this pre requisite approach of handling grpc and instead used rebar prehooks.  Be worth checking that setup out.  Keeps things cleaner and more maintainable

Reply:
Apparently, one side-effect of using rebar3's `pre_hooks` is that grpc modules get regenerated too frequently.

We originally had `pre_hooks` consistent with miner in the first commit of this branch but requested to be removed from `rebar.conf`, so then `make compile` for full builds and `rebar compile` for rapid iterations.

Is there a flaw in that reasoning?